specreduce/core.py

```diff
@@ -2,3 +2,246 @@
+        # doctest: +SKIP
+
+        If no std is supplied, will raise a ValueError.
+
+    mode : str, optional
+        Default to 'spline'. Can also be set to either 'linear' or 'poly'.
+
+    polyorder: float, optional
+        Default to '6'. Is defined as the order of the polynomial for the 'poly' fitting.
+        Only needed if the 'poly' mode has been chosen.
+
+    show: bool, optional
+        Default to 'False'. If set to 'True', shows a plot of the sensfunc determined.
+
+    Returns
+    -------
+    sensfunc : a 1D numpy array
+        The sensfunc derived with the standard star for the flux calibration.
+
+    """
+
+    if std is None:
+        msg1 = "Error: No 'onedstd' standard star file has been specified." \
+               "Cannot proceed further."
+        raise ValueError(msg1)
+
+    std = std.lower()
+    std_dir = os.path.join(os.path.dirname(os.path.realpath(__file__)),
+                           "datasets", "onedstds")
+
+    if os.path.isfile(os.path.join(std_dir, std)):
+        std_wave, std_mag, std_wth = np.genfromtxt(
+            os.path.join(std_dir, std), skip_header=1, unpack=True
+        )
+
+        # Convert standard star from magnitude units to flux
+        std_flux = _mag2flux(std_wave, std_mag)
+
+        # Define an array of the three most prominent Balmer lines
+        balmer = np.array([6563, 4861, 4341], dtype="float")
+
+        # Down-sample (coded as ds) the observed flux to fit the standard star's bins
+        obj_flux_ds = []
+        obj_wave_ds = []
+        std_flux_ds = []
+
+        for i in range(len(std_wave)):
+            rng = np.where((obj_wave >= std_wave[i] - std_wth[i] / 2.0) &
+                           (obj_wave < std_wave[i] + std_wth[i] / 2.0))
+            isbalmer = np.where((balmer >= std_wave[i] - std_wth[i] / 2.0) &
+                                (balmer < std_wave[i] + std_wth[i] / 2.0))
+
+            # Check to see if the bin does indeed contain observed spectra,
+            # with no presence of the Balmer lines defined previously
+            if (len(rng[0]) > 1) and (len(isbalmer[0]) == 0):
+                obj_flux_ds.append(np.nanmean(obj_flux[rng]))
+                obj_wave_ds.append(std_wave[i])
+                std_flux_ds.append(std_flux[i])
+
+        # To ensure the ratio between the standard star flux and observed flux
+        # has proper flux units
+        ratio = np.abs(np.array(std_flux_ds, dtype="float") /
+                       np.array(obj_flux_ds, dtype="float"))
+
+        # Interpolate flux calibration (sensfunc) onto object's wavelength grid
+        # 3 modes of interpolation available: 'linear', 'cubic spline', and 'polynomial'
+
+        # To check if a mode of interpolation has been specified
+        if mode not in ("linear", "spline", "poly"):
+            mode = "spline"
+            print("Warning: No interpolation mode has been set"
```

Comment:
Use python internal warning rather than print functions so that users can set the warning levels.

Reply:
Okay sure, I could do something like the following: 

```python
import warnings

warnings.warn('No interpolation mode has been set.')
```